packages/browser/coretest/__tests__/StitchAppClientIntTests.ts

```diff
@@ -330,4 +330,116 @@ describe("StitchAppClient", () => {
     expect(resultDoc["intValue"]).toEqual(42);
     expect(resultDoc["stringValue"]).toEqual("hello");
   });
+
+  it("should update lastAuthActivity on various auth events", async () => {
+    const [appResponse, app] = await harness.createApp();
+    await harness.addProvider(app as App, new Anon());
+    await harness.addProvider(
+      app as App,
+      new Userpass(
+        "http://emailConfirmUrl.com",
+        "http://resetPasswordUrl.com",
+        "email subject",
+        "password subject"
+      )
+    );
+
+    let client = harness.getAppClient(appResponse as AppResponse);
+
+    const checkpoint1 = new Date();
+
+    // login with email provider
+    const user1Id = await harness.registerAndLoginWithUserPass(
+      app as App,
+      client,
+      "test@10gen.com",
+      "hunter1"
+    );
+
+    // login should set an authActivity after the pre-login time
+    expect(client.auth.user!.lastAuthActivity.getTime()).toBeGreaterThan(
```

Comment:
I wanted to avoid equality checks, because that will mask problems where the lastAuthActivity doesn't update at all. In most checks, the new time is after an `await` which almost always takes a few ms. The one place where we didn't have an await, I added an artificial wait.